### ISSUE TITLE
Create a Test Project that Validates the Functionality of the Reusable GitHub Actions

### ISSUE THREAD

**Issue by PSchmiedmayer** [StanfordBDHG/.github#19]
### Problem

- The current repo setup relies on external tests in other repos to validate the functionality of the build scripts 

### Solution

- The repo should contain a test project that contains multiple targets to validate the functionality of the different GitHub Actions

### Additional context

This issue is a great way to get started with contributing to the GitHub Actions and getting familiar with the issue as it mainly consists of testing and validating the existing functionality.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct and Contributing Guidelines

**Comment by Supereg** [comment]
Wouldn’t it make sense to include the project inside this repository to allow for automated PR testing such that everything is self-contained?
Similar to how we do it with the swiftlint action?

**Comment by PSchmiedmayer** [comment]
@Supereg Yes, that was the intention of the issue, including it in this project is the way to go 👍

**Comment by PSchmiedmayer** [comment]
https://github.com/StanfordBDHG/.github/pull/41 provides an other motivation for this, we should see that we can also catch regressions in these workflows.
